### PR TITLE
Add namespaced trusted tool registry

### LINKED ISSUE #7: Replace global escape codes with a namespaced tool registry
Move from the global escape_codes dictionary toward a namespaced tool registry that can support trusted preload tools, future role-owned tools, and safer execution metadata.\n\nAcceptance criteria:\n- Add a registry abstraction with namespace/name lookup, argument metadata, and callable execution.\n- Migrate create_role preload behavior onto the registry without changing user-visible behavior.\n- Keep trusted tool definition loading separate from untrusted model output.\n- Add tests for duplicate names, missing args, invalid identifiers, and namespace lookup.\n- Document the expected preload format or compatibility path.

### PR DESCRIPTION
**Pull request by displague**
## Summary
- replace the legacy preload path with a trusted, namespaced tool registry backed by `tools.yaml`
- expose tool metadata for OpenAI-compatible Responses function tools and chat-completions function tools, including reverse lookup for model-facing tool names
- update runtime docs, AGENTS guidance, and repo-local skills around trusted tools, operator scope, and future scheduling direction

## Validation
- `C:\Users\displ\Documents\robits\.venv\Scripts\python.exe -m unittest` (19 tests)
- `C:\Users\displ\Documents\robits\.venv\Scripts\python.exe -m py_compile main.py tests\test_runtime.py`
- `C:\Users\displ\Documents\robits\.venv\Scripts\python.exe -c "import yaml; yaml.safe_load(open('tools.yaml', encoding='utf-8')); print('tools yaml ok')"`
- `python C:\Users\displ\.codex\skills\.system\skill-creator\scripts\quick_validate.py .github\skills\robits-runtime`
- `git diff --check`
- bounded OpenAI-compatible local smoke run with `main.py --prompt "HR, say hello in one short sentence." --turns 1`

## Review Follow-up
- Addressed all 5 Copilot comments in 433f165 and replied on each thread.
- Resolved all 5 review threads after patching.

Acting as Codex GPT-5.5 on behalf of displague.

Closes #7

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -47,6 +47,175 @@ def make_client():
...
+    description: str
+    parameters: dict
+    args: list[str]
+    func: object
+    aliases: tuple[str, ...] = ()
+
+    @property
+    def openai_name(self):
+        return self.name.replace(".", "__")
+
```
[the remark this answers is not in the document]
Comment (displague) [reply main.py:63]: Addressed in 433f165 by automatically registering each tool's OpenAI-safe name (for example, `org__create_role`) as a reverse alias to the canonical namespaced tool. Added `test_openai_tool_name_exec_resolves_tool` to verify model-facing function-call names execute correctly.

Acting as Codex GPT-5.5 on behalf of displague.

File: main.py [hunk 2]
```diff
@@ -47,6 +47,175 @@ def make_client():
...
+        required = parameters.get("required", [])
+        properties = parameters.get("properties", {})
+        if not isinstance(properties, dict) or not isinstance(required, list):
+            raise ValueError("Tool parameters must contain object properties and required list.")
+        arg_names = []
+        for arg_name in required:
+            if not isinstance(arg_name, str) or arg_name not in properties:
+                raise ValueError("Tool required args must be named properties.")
+            arg_names.append(arg_name)
```
[the remark this answers is not in the document]
Comment (displague) [reply main.py]: Addressed in 433f165 by compiling all declared parameter properties, tracking required args separately, defaulting optional parameters to `None`, and rejecting unexpected call args. Added `test_optional_tool_property_can_be_supplied_or_omitted` for both optional-property paths.

Acting as Codex GPT-5.5 on behalf of displague.

File: main.py [hunk 3]
```diff
@@ -47,6 +47,175 @@ def make_client():
...
+
+        return name, description, parameters, arg_names, code, aliases
+
+    def register_definition(self, instruction):
+        name, description, parameters, arg_names, code, aliases = self.normalize_definition(instruction)
+        if name in self._tools:
+            raise ValueError(f"Tool '{name}' already exists.")
+        for alias in aliases:
```
[the remark this answers is not in the document]
Comment (displague) [reply main.py:188]: Addressed in 433f165 by validating every explicit alias with the same `validate_tool_name()` path before registration. Added `test_invalid_tool_alias_is_rejected` so bad aliases fail at load time instead of becoming uncallable later.

Acting as Codex GPT-5.5 on behalf of displague.

File: main.py [hunk 4]
```diff
@@ -267,17 +385,15 @@ def parse_escape_code(s):
     return None
 
 
-def preload(system, yaml_file_path=None):
-    yaml_file_path = yaml_file_path or Path(__file__).with_name("preload.yaml")
+def load_tools(system, yaml_file_path=None):
+    yaml_file_path = yaml_file_path or Path(__file__).with_name("tools.yaml")
     with open(yaml_file_path, 'r') as file:
         yaml_content = yaml.safe_load(file)
```
[the remark this answers is not in the document]
Comment (displague) [reply main.py]: Addressed in 433f165 by opening tool files with `encoding="utf-8"` and raising a clear `ValueError` when the YAML root is not a list. Added `test_load_tools_requires_list_file` for the invalid-shape case.

Acting as Codex GPT-5.5 on behalf of displague.

File: tests/test_runtime.py [hunk 5]
```diff
@@ -202,6 +202,91 @@ def test_escape_code_builtins_are_restricted(self):
...
+            main.load_tools(system)
+
+        tools = main.tool_registry.as_chat_completion_tools()
+
+        self.assertEqual(tools[0]["type"], "function")
+        self.assertEqual(tools[0]["function"]["name"], "org__create_role")
+        self.assertEqual(
+            tools[0]["function"]["parameters"]["required"],
+            ["role_name", "role_description"],
+        )
```
[the remark this answers is not in the document]
Comment (displague) [reply tests/test_runtime.py:230]: Addressed in 433f165 by changing both metadata tests to search for the `org__create_role` entry by name instead of assuming index `[0]`.

Acting as Codex GPT-5.5 on behalf of displague.
